Cabal/Distribution/Version.hs

```diff
@@ -805,3 +807,29 @@ instance Text VersionRange where
                      (">",  LaterVersion),
                      (">=", orLaterVersion),
                      ("==", ThisVersion) ]
+
+-- | Does the version range have an upper bound?
+--
+-- @since 1.24.0.0
+hasUpperBound :: VersionRange -> Bool
+hasUpperBound AnyVersion = False
+hasUpperBound (ThisVersion _) = True
+hasUpperBound (LaterVersion _) = False
+hasUpperBound (EarlierVersion _) = True
+hasUpperBound (WildcardVersion _) = True
+hasUpperBound (UnionVersionRanges x y) = hasUpperBound x && hasUpperBound y
+hasUpperBound (IntersectVersionRanges x y) = hasUpperBound x && hasUpperBound y
```

Comment:
I don't think that this is correct. Consider the intersection of `>=1.0` with `<=1.1`.